controllers/root.go

```diff
@@ -34,7 +34,8 @@ func (c *Context) Ping(rw web.ResponseWriter, req *web.Request) {
 func (c *Context) LoginHandshake(rw web.ResponseWriter, req *web.Request) {
 	if token := helpers.GetValidToken(req.Request, c.Settings); token != nil {
 		// We should just go to dashboard if the user already has a valid token.
-		http.Redirect(rw, req.Request, "/#/dashboard", http.StatusFound)
+		dashboardURL := fmt.Sprintf("%s%s", c.Settings.AppURL, "/#/dashboard")
```

Comment:
It's not needed, I prefer to be strict about what we send the client. It's the whole robustness principle "Be conservative in what you do, be liberal in what you accept from others". Unless there's a feature where we're supporting a whitelist of multiple allowed hosts, I think we should just use absolute URLs.